spec/facter/facts/solaris/is_virtual_spec.rb

```diff
@@ -25,16 +26,33 @@
         allow(Facter::Resolvers::Solaris::Dmi).to receive(:resolve).with(:product_name).and_return('unkown')
         allow(Facter::Resolvers::Solaris::Dmi).to receive(:resolve).with(:bios_vendor).and_return('unkown')
         allow(Facter::Resolvers::VirtWhat).to receive(:resolve).with(:vm).and_return(nil)
+        allow(Facter::Resolvers::Xen).to receive(:resolve).with(:vm).and_return(nil)
       end
 
-      it 'returns is_virtual fact as physical' do
+      it 'returns is_virtual fact as false' do
```

Comment:
The fact should never return `false`. In worst case, it does not find any hypervisor and returns `physical `.  It would be better to change `vm` to `physical`

Reply:
is_virtual returns true or false. virtual may return physical